sdks/go/pkg/beam/core/runtime/harness/session.go

```diff
@@ -58,21 +56,12 @@ func setupDiagnosticRecording() error {
 		return nil
 	}
 
-	var err error
-
-	storagePath = runtime.GlobalOptions.Get("storage_path")
-	// Any form of recording requires the destination directory to exist.
-	if err = os.MkdirAll(storagePath, 0755); err != nil {
-		return fmt.Errorf("Unable to create session directory: %v", err)
-	}
-
 	if !isEnabled("session_recording") {
```

Comment:
(In connection with the high-level comment above)

This kind of check shouldn't be needed here. A hook is enabled iff it's invoked.